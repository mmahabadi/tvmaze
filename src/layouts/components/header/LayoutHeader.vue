<template>
  <header>
    <div class="container">
      <div class="col-span-1 lg:block hidden"><slot name="logo"></slot></div>

      <div class="col-span-3">
        <slot name="search"></slot>
      </div>
      <div class="header-inner">
        <slot></slot>
      </div>
    </div>
  </header>
</template>
<style>
header {
  @apply shadow-md sticky top-0 z-20 backdrop-blur-sm bg-gray-900/30;

  .container {
    @apply mx-auto py-6 px-2 lg:grid gap-10 grid-cols-7 justify-between items-center;

    .header-inner {
      @apply col-span-3 font-medium text-sm hidden xl:gap-14 2xl:gap-20 justify-between lg:flex xl:justify-end items-center;
    }
  }
}
</style>
