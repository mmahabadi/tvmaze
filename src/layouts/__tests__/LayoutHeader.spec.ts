import { mount, type VueWrapper } from '@vue/test-utils'
import { describe, expect, beforeEach, it } from 'vitest'
import LayoutHeader from '@/layouts/components/header/LayoutHeader.vue'

describe('LayoutHeader', () => {
  let wrapper: VueWrapper<LayoutHeader>

  beforeEach(() => {
    wrapper = mount(LayoutHeader)
  })

  it('renders correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
