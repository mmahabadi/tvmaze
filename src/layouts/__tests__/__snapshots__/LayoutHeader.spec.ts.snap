// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LayoutHeader > renders correctly 1`] = `
"<header>
  <div class="container">
    <div class="col-span-1 lg:block hidden"></div>
    <div class="col-span-3"></div>
    <div class="header-inner"></div>
  </div>
</header>"
`;
